crates/red_knot_workspace/src/workspace/metadata.rs

```diff
@@ -78,6 +202,33 @@ impl WorkspaceMetadata {
 }
 
 impl PackageMetadata {
+    pub(crate) fn from_pyproject(
+        pyproject: PyProject,
+        root: SystemPathBuf,
+        base_configuration: Option<&Configuration>,
+    ) -> Result<Self, WorkspaceDiscoveryError> {
+        let Some(project) = pyproject.project else {
+            return Err(WorkspaceDiscoveryError::MissingProjectTable { package_path: root });
+        };
+
+        let Some(name) = project.name.as_ref() else {
+            return Err(WorkspaceDiscoveryError::MissingPackageName { package_path: root });
```

Comment:
I don't think there's a strong reason for it. The same as that it doesn't matter much to us whether packages have unique names inside a workspace. But I do see value in having the discovery consistent between uv and red-knot and `name` is a required setting according to the [`pyproject.toml` specification](https://packaging.python.org/en/latest/specifications/pyproject-toml/#declaring-project-metadata-the-project-table), unless the `project` table isn't specified at all. 

That's why I think the question rather is: Should the `project` table be required?

Reply:
> Is there any reason this needs to be an error for us? It seems odd to me that we'll happily synthesize a workspace with the name <virtual> 

We also only use `virtual` in case red knot runs in `/`. It otherwise uses the directory name as fallback

Reply:
> I'm leaning towards keeping it as is for now. We might actually need the name, depending on where we land on how workspaces integrate into module resolution (do we rely on the packager, in this case we won't need the name, or do we have our own, custom handling for packages, in which case we need the name)

I don't understand how these two things are related at all, or why we would ever need a project name for module resolution. The project name is just PyPI metadata that's unrelated to the import system or package names.

Reply:
We talked this through in our 1:1 and this was a misunderstanding on my end of how python does module resolution.

Reply:
I changed it to make the project table and name optional